src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -192,28 +192,42 @@ private static class \{holderClass} {
         } else {
             String invokerClassName = newHolderClassName(javaName);
             String paramExprs = paramExprs(declType, finalParamNames, isVarArg);
-            String varargsParam = finalParamNames.get(finalParamNames.size() - 1);
             appendBlankLine();
             emitDocComment(decl, "Variadic invoker interface for:");
             appendLines(STR."""
                 public interface \{invokerClassName} {
+                    FunctionDescriptor BASE_DESC = \{functionDescriptorString(2, decl.type())};
```

Comment:
My 0.02$ (related to your question 1): let's make this a record which takes a method handle and a specialized descriptor, which are then exposed with the record accessors, as usual. Drop the static accessors, after all, a client can access the specialized MH/desc by first creating an instance of this invoker class (which will be likely escaped away if the client only wants the MH). Then, we can hide the non-specialized fields as `private`.

Reply:
Good idea